### PR TITLE
Add optional info function to redis_module macro

### PR DESCRIPTION
**Pull request by oshadmi**
Adding a new argument to `redis_module` macro, which allows specifying a callback function for adding specific module information to the `INFO` command,
e.g., `INFO MODULES`,
or `INFO <module_name>`, for example:
```
127.0.0.1:6379> INFO hello
# hello_hello
hello_field:value
127.0.0.1:6379> 

```

(this is in addition to the already used callback function for adding stack trace on panic/crash)

### CODE REVIEW

File: src/context/mod.rs [hunk 1]
```diff
@@ -286,3 +286,36 @@ impl Context {
         unsafe { raw::RedisModule_SetModuleOptions.unwrap()(self.ctx, options.bits()) };
     }
 }
+
+pub struct InfoContext {
+    pub ctx: *mut raw::RedisModuleInfoCtx,
```
Comment (gkorland) [inline src/context/mod.rs:291]: @oshadmi why `pub`?

File: src/macros.rs [hunk 2]
```diff
@@ -113,7 +114,10 @@ macro_rules! redis_module {
             ctx: *mut $crate::raw::RedisModuleInfoCtx,
             for_crash_report: i32,
         ) {
-            $crate::base_info_func(ctx, for_crash_report == 1);
+            use $crate::InfoContext;
+            let mut __info_func__cb : Option<fn(&InfoContext, bool)> = None;
+            $( __info_func__cb = Some($info_func); )?
+            $crate::base_info_func(&$crate::InfoContext::new(ctx), for_crash_report == 1, __info_func__cb);
```
Comment (gkorland) [inline src/macros.rs:120]: @oshadmi no issues calling `base_info_func` when `__info_func__cb==None`?
Do we need to call `base_info_func` method if no `info_func` was passed?

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/context/mod.rs (hunk 1 above)
Comment (oshadmi) [reply src/context/mod.rs:291]: It is used outside, similar to `Context`, e.g., used by other modules, such as RedisJSON, etc.

File: src/macros.rs (hunk 2 above)
Comment (oshadmi) [reply src/macros.rs:120]: We need `base_info_func` to add panic/crash info - always.
The user info is optional, and when it is `None`, no info is added, except for panic/crash info.

**Review by gkorland — APPROVED** [review]

**Pull request merged**